chapters/scoping.tex

```diff
@@ -55,7 +55,7 @@ \section{Static Name Lookup}\label{static-name-lookup}
 \subsection{Simple Name Lookup}\label{simple-name-lookup}
 
 % Not adding 'encapsulated' as normal word to index; the keyword variant seems sufficient
-A class declared with the keyword \lstinline!encapsulated!\indexinline{encapsulated} (see \lstinline[language=grammar]!class-definition! in the grammar) is called an \firstuse{encapsulated} class.
+A class declared with the keyword \lstinline!encapsulated!\indexinline{encapsulated} (see \lstinline[language=grammar]!class-definition! in the grammar) is called an \firstuse[---]{encapsulated} class.
```

Comment:
That shows up as "is called an encapsulatedNOTHING class." in [the Jenkins build](https://test.openmodelica.org/jenkins/job/ModelicaAssociation/job/ModelicaSpecification/view/change-requests/job/PR-3043/).

Reply:
Same as above.

Reply:
Fixed.

Reply:
The reason in this case is that we already have `\indexinline{encapsulated}` in the same paragraph, and it seemed a bit excessive to both have entries for the keyword and the natural language word pointing to the same paragraph.  Don't you think?

Reply:
Oh, sorry. I missed that there were three entries in the index (HTML):
![image](https://user-images.githubusercontent.com/20460714/143041065-e6322e62-5b71-4d85-9254-402c0e88e0cd.png)
Actually, in PDF there are only two entries:
![image](https://user-images.githubusercontent.com/20460714/143043954-461d0d9d-aa1d-4343-a892-e1cabfab0d2d.png)
I guess two of them on the same page, so they got merged?

Right, I first looked at the pdf and tried to follow the links there. But the pdf shows the whole page when you click on the link, and I wanted to see the exact place. So I switched to the HTML. I clicked on the first two links because in my mind there were only two, and missed the third one completely.

Is there a way to sort the entries in the index? The first item on that list is in Section 5.3.2, the second in  5,3,3, and the third in 5.3.1 is this paragraph above. It would make more sense to have 5.3.1 to be the first one.

Reply:
The discussion got moved to the Converstation tab. This is not really a problem with the PR, so resolving.